### PR TITLE
ds-index: Add a BIOS dumping guide

### PR DESCRIPTION
**Pull request by lifehackerhansol**

### CODE REVIEW

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+1. Download [dsibiosdumper](https://melonds.kuribo64.net/downloads/dsibiosdumper.7z)
+1. Copy `dsibiosdumper.nds` from the dsibiosdumper `.7z` file to the root of your SD card
+1. Reinsert your SD card into your device
+
+### Part 2: dsibiosdumper
+1. Power on your device
+1. Launch TWiLight Menu++
+1. Navigate to the root of your SD card, and launch dsibiosdumper
+1. Press (A) to dump everything
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:35]:
```suggestion
1. Press <kbd class="face">A</kbd> to dump everything
```

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
+lang: en-US
+layout: wiki
+section: ds-index
+category: guides
+title: BIOS / Firmware Dumping
+description: How to extract the Nintendo DS or DSi BIOS / Firmware from your console
+tabs:
+  - tab-twl-sd-card: DSi / 3DS with custom firmware
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:9]:
```suggestion
title: BIOS/Firmware Dumping
description: How to extract the Nintendo DS or DSi BIOS/Firmware from your console
tabs:
  - tab-twl-sd-card: DSi/3DS with custom firmware
```

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    tab-flashcard: Flashcard
+---
+
+Some emulators, such as MelonDS, require the DS-mode or DSi-mode BIOS and firmware, which is only available by extracting from your console. There are two distinct ways to achieve this, being split into a DSi or 3DS with custom firmware, or a DS / DS Lite / DSi or 3DS without custom firmware and a flashcard compatible with the console.
+
+The DSi / 3DS method will extract both the DS-mode and DSi-mode BIOS and firmware files. The Flashcard method will only extract the DS-mode files. It is highly recommended to use the DSi / 3DS method where possible.
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:15]:
```suggestion
The DSi/3DS method will extract both the DS-mode and DSi-mode BIOS and firmware files. The flashcard method will only extract the DS-mode files. It is highly recommended to use the DSi/3DS method where possible.
```

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+1. Reinsert your SD card into your device
+
+### Part 2: dsibiosdumper
+1. Power on your device
+1. Launch TWiLight Menu++
+1. Navigate to the root of your SD card, and launch dsibiosdumper
+1. Press (A) to dump everything
+1. Wait for the process to complete
+    - This can take a while
+1. After the process is complete, press (Select) to exit dsibiosdumper
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:38]:
```suggestion
1. After the process is complete, press <kbd>SELECT</kbd> to exit dsibiosdumper
```

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+1. Download [dsbf_dump](https://github.com/DS-Homebrew/dsbf_dump/releases/latest) (the `.nds` file)
+1. Copy `dsbf_dump.nds` to the root of your SD card
+1. Reinsert your SD card into your flashcard
+
+### Part 2: dsbf_dump
+1. Power on your device
+1. Launch your flashcard
+1. Navigate to the root of your SD card, and launch dsbf_dump
+1. Wait for the process to complete
+1. After the process is complete, press (Start) to power off your device
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:64]:
```suggestion
1. After the process is complete, press <kbd>START</kbd> to power off your device
```

**Review by Epicpkmn11 — CHANGES_REQUESTED** [review]
Formatting suggestions

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md (hunk 3 above)
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:15]: The 3DS cannot extract full DSi mode BIOS files, at best they *might* work for some things, but they will be the incomplete dumps missing RSA keys

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+### Part 1: Getting the required files
+
+1. Download [dsibiosdumper](https://melonds.kuribo64.net/downloads/dsibiosdumper.7z)
+1. Copy `dsibiosdumper.nds` from the dsibiosdumper `.7z` file to the root of your SD card
+1. Reinsert your SD card into your device
+
+### Part 2: dsibiosdumper
+1. Power on your device
+1. Launch TWiLight Menu++
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:33]: dsibiosdumper should always be run from Unlaunch if at all possible, dumps from TWiLight Menu++ will be incomplete same as dumps made on 3DS

**Review by Epicpkmn11 — CHANGES_REQUESTED** [review]

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md (hunk 6 above)
Comment (lifehackerhansol) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md:33]: How do you suggest we run this on a 3DS? It should work there, but we don't really have a valid way to run it other than some sorta homebrew launcher.

Then again, maybe it doesn't support dumping NAND, and thus would need a separate guide...

~~another tool maybe?~~

Comment (Epicpkmn11) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md:33]: TWiLight, but either 3DS should just be left out as a bad idea or separate from DSi

Comment (lifehackerhansol) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md:33]: does it work as a cia? lol

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md (hunk 3 above)
Comment (lifehackerhansol) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md]: No longer using dsibiosdumper on 3DS

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md (hunk 6 above)
Comment (Epicpkmn11) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md]: it wouldn’t help

File: pages/_en-US/ds-index/ds-bios-firmware-dump.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+section: ds-index
+category: guides
+title: BIOS/Firmware Dumping
+description: How to extract the Nintendo DS or DSi BIOS/Firmware from your console
+tabs:
+  - tab-dsi-sd-card: DSi with custom firmware
+    tab-flashcard: Flashcard
+---
+
+Some emulators, such as melonDS, require the DS-mode or DSi-mode BIOS and firmware, which is only available by extracting from your console. There are two distinct ways to achieve this, being split into a DSi with custom firmware, or a DS, DS Lite, DSi or 3DS without custom firmware and a flashcard compatible with the console.
```
Comment (Epicpkmn11) [inline pages/_en-US/ds-index/ds-bios-firmware-dump.md:13]:
```suggestion
Some emulators, such as melonDS, require the DS-mode or DSi-mode BIOS and firmware, which are only available by extracting from your console. There are two distinct ways to achieve this, being split into a DSi with custom firmware, or a DS, DS Lite, DSi, or 3DS without custom firmware and a flashcard compatible with the console.
```
couple more 😅

Comment (Epicpkmn11) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md:13]: also thinking on it i’m not sure “only available” is the best thing to say, that implies they’re console specific, perhaps “available by” would be better?

Comment (lifehackerhansol) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md:13]: I am failing to understand that point. It _is_ console-specific; you can't obtain them without a console, no? Unless we're just fully endorsing piracy now

Comment (Epicpkmn11) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md:13]: console specific means unique to one console, I'm not endorsing it, but I don't like incorrectly stating things don't exist

Comment (lifehackerhansol) [reply pages/_en-US/ds-index/ds-bios-firmware-dump.md]: All points addressed; used a list instead of whatever jargon that was earlier

**Pull request merged**